src/app/ubs-user/ubs-user-messages/ubs-user-messages.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, OnInit } from '@angular/core';
+import { UserMessagesService } from '../services/user-messages.service';
+import { LocalStorageService } from '@global-service/localstorage/local-storage.service';
+import { NotificationBody } from '../../ubs-admin/models/ubs-user.model';
+import { ActivatedRoute, Router } from '@angular/router';
+
+@Component({
+  selector: 'app-ubs-user-messages',
+  templateUrl: './ubs-user-messages.component.html',
+  styleUrls: ['./ubs-user-messages.component.scss']
+})
+export class UbsUserMessagesComponent implements OnInit {
+  isAnyMessages = true;
+  notifications: NotificationBody[];
+  panelOpenState = false;
+  page = 1;
+  count = 0;
+  pageSize = 10;
+  language = this.localStorage.getCurrentLanguage();
+  isLoadSpinner: boolean;
+  isLoadSmallSpinner: boolean;
+  isLoadBar: boolean;
+  public countOfMessages: number;
+  localization = {
+    title: 'ubs-user-notification.title',
+    id: 'ubs-user-notification.title-table.number',
+    themeMessages: 'ubs-user-notification.title-table.theme-messages',
+    time: 'ubs-user-notification.title-table.time'
+  };
+
+  constructor(
+    private userMessagesService: UserMessagesService,
+    private localStorage: LocalStorageService,
+    private router: Router,
+    private route: ActivatedRoute
+  ) {}
+
+  ngOnInit() {
+    this.language = this.localStorage.getCurrentLanguage();
+    this.isLoadSpinner = true;
+    this.route.params.subscribe((val) => {
+      this.page = +this.route.snapshot.paramMap.get('pageId');
+      this.fetchNotification();
+    });
+  }
+
+  fetchNotification(): void {
+    this.userMessagesService.getNotification(this.language, this.page - 1, this.pageSize).subscribe(
```

Comment:
add takeUntil() and unsubscribe in ngDestroy or take()

Reply:
Done